### PR TITLE
cloud watch changes - working version

### PR DESCRIPTION
**Pull request by pmoogi-redhat**
### Description

Conf generation for Vector supporting logs forwarding to cloudwatch
+ unit test case testing for this vector configuration

- Depending on PR(s):
- Bugzilla:
- Github issue:
- JIRA: https://issues.redhat.com/browse/LOG-2223
- Enhancement proposal:

/cc @vimalk78

### CODE REVIEW

File: internal/generator/vector/elements/elements.go [hunk 1]
```diff
@@ -31,10 +31,12 @@ route.{{$route_name}} = {{$route_expr}}
...
 type Remap struct {
-	ComponentID string
-	Desc        string
-	Inputs      string
-	VRL         string
+	ComponentID    string
+	Desc           string
+	Inputs         string
+	GroupBy        string
+	LogGroupPrefix string
```
Comment (vimalk78) [inline internal/generator/vector/elements/elements.go:38]: strongly suggest not to change the basic elements for a specific purpose.
vector Remap does not have these fields https://vector.dev/docs/reference/configuration/transforms/remap/

File: internal/generator/vector/source/tags.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package source
+
+const (
+	ApplicationTags    = "kubernetes.**"
+	JournalTags        = "journal.** system.var.log**"
+	InfraContainerTags = "kubernetes.var.log.pods.openshift-*_** kubernetes.var.log.pods.default_** kubernetes.var.log.pods.kube-*_**"
+	InfraTags          = InfraContainerTags + " " + JournalTags
+	AuditTags          = "linux-audit.log** k8s-audit.log** openshift-audit.log** ovn-audit.log**"
```
Comment (vimalk78) [inline internal/generator/vector/source/tags.go:8]: where are these values used?

File: internal/generator/vector/source/audit_logs.go [hunk 3]
```diff
@@ -9,6 +9,8 @@ const HostAuditLogTemplate = `
 # {{.Desc}}
 [sources.{{.ComponentID}}]
 type = "file"
+file_key = "file"
+host_key = "host"
```
Comment (vimalk78) [inline internal/generator/vector/source/audit_logs.go:13]: where is `file` key used? `host` seems to be used in cloudwatch VRL

File: internal/generator/vector/output/cloudwatch/cloudwatch.go [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+		Cw_groupName_infra = "unknown"
+		Cw_streamName_infra = "unknown"
+		Cw_groupName_audit = "unknown"
+		Cw_streamName_audit = "unknown"
+
+	if .GroupBy == "logType" {
+		Cw_groupName_app = "application"
+		Cw_streamName_app = "{{.ComponentID}}"
+		Cw_groupName_infra = "infrastructure"
+		Cw_streamName_infra = join!([{{.host}},".",{{.ComponentID}}])
```
Comment (vimalk78) [inline internal/generator/vector/output/cloudwatch/cloudwatch.go:79]: suggest not to use component id in the VRL. component ID is just used to name a source/sink/transform in vector config.

File: test/functional/outputs/cloudwatch/forward_to_cloudwatch_test.go [hunk 5]
```diff
@@ -64,7 +64,7 @@ var _ = Describe("[Functional][Outputs][CloudWatch] Forward Output to CloudWatch
 		defaultRetryInterval, _ = time.ParseDuration("10s")
 
 		getRawCloudwatchApplicationLogs = func(f *functional.CollectorFunctionalFramework, cwlClient *cwl.Client) ([]string, error) {
-			log.V(3).Info("Retrieving cloudwatch LogGroups")
+			log.V(2).Info("Retrieving cloudwatch LogGroups")
```
Comment (vimalk78) [inline test/functional/outputs/cloudwatch/forward_to_cloudwatch_test.go:67]: is this functional test enabled for vector?

File: internal/generator/vector/output/cloudwatch/cloudwatch.go [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+	.Cw_streamName = Cw_streamName_app
+	}
+	if .log_level == "infrastructure" {
+	.Cw_groupName = Cw_groupName_infra 
+	.Cw_streamName = Cw_streamName_infra
+	}
+	if .log_level == "audit" {
+	.Cw_groupName = Cw_groupName_audit
+	.Cw_streamName = Cw_streamName_audit
+	}
```
Comment (vimalk78) [inline internal/generator/vector/output/cloudwatch/cloudwatch.go:110]: where is `.log_level` declared? did you mean `.log_type` ?

**Review by vimalk78 — CHANGES_REQUESTED** [review]
suggested changes

File: internal/generator/vector/source/audit_logs.go (hunk 3 above)
Comment (pmoogi-redhat) [reply internal/generator/vector/source/audit_logs.go]: yes host is required for cloudwatch but file_key is not required so taking it out.

File: test/functional/outputs/cloudwatch/forward_to_cloudwatch_test.go (hunk 5 above)
Comment (vimalk78) [reply test/functional/outputs/cloudwatch/forward_to_cloudwatch_test.go:67]: this variable only setup the collector as vector. this variable does not choose which tests will run under `make test-functional-vector`

**Pull request closed without merging**